### DISCUSSION TITLE
check during an entry if the email sent is already used

### DISCUSSION THREAD

**Discussion by CogalTek** [pocketbase/pocketbase#3558] [Q&A] [answered]
Hello,

I would like to check during an entry if the email sent is already in use and if this is the case return an error message, could you ask if this can be managed via the API? or should this be done through the web app?

**Comment by ganigeorgiev** [comment, marked as answer]
Sorry, I'm not sure that I understand what do you mean. Could you elaborate on your use case and/or provide an example of what you are trying to do?

If you are asking for a unique constraint you can create it from the Admin UI in the Collections edit panel under the "Unique constraints and indexes" section:

![indexes_popup](https://camo.githubusercontent.com/c5cbf751c9a05c5320d871bf4747134b64eda31b7880ccf9fe997f13fbd5dad9/68747470733a2f2f692e696d6775722e636f6d2f5556796f3065662e706e67)

For the auth collections, this is defined out of the box for the `username` and `email` fields.

**Reply by CogalTek** [reply]
I tested the solution and it worked, thank you

**Comment by CogalTek** [comment]
![Capture d’écran 2023-10-19 à 17 42 08](https://github.com/pocketbase/pocketbase/assets/91875582/15eab19c-4597-4e4a-a040-ce8af781caa7)

I'm trying to ensure that if someone tries to register with an email address but it's already taken, it won't accept it and if it's possible to return a special error code

**Reply by ganigeorgiev** [reply]
>  if someone tries to register with an email address

If you use an [auth collection](https://pocketbase.io/docs/collections/#auth-collection), this checks are done automatically for you so you don't have to do anything.

If you want a unique constraint for your base collection field, Go to the "Fields tab > Unique constraints and indexes" and create a **Unique** index for it, you don't have to list it in the rules.

**Reply by ganigeorgiev** [reply]
Also you can remove the empty checks from the rule and mark your fields as "Nonempty" if thats the requirement you want:
![field_options](https://i.imgur.com/kWaPAil.png)

**Reply by CogalTek** [reply]
thank's
